src/main/java/com/angelbroking/smartapi/SmartConnect.java

```diff
@@ -1,727 +1,781 @@
-	 * 
-	 * @param status is list of gtt rule status.
-	 * @param page   is no of page
-	 * @param count  is the count of gtt rules
-	 * @return returns the detailed list of gtt rules.
-	 */
-	public JSONArray gttRuleList(List<String> status, Integer page, Integer count) {
-		try {
-			JSONObject params = new JSONObject();
-			params.put("status", status);
-			params.put("page", page);
-			params.put("count", count);
-
-			String url = routes.get("api.gtt.list");
-			JSONObject response = smartAPIRequestHandler.postRequest(this.apiKey, url, params, accessToken);
-			System.out.println(response);
-			return response.getJSONArray("data");
-		} catch (Exception | SmartAPIException e) {
-			System.out.println(e.getMessage());
-			return null;
-		}
-
-	}
-
-	/**
-	 * Get Historic Data.
-	 * 
-	 * @param params is historic data params.
-	 * @return returns the details of historic data.
-	 */
-	public String candleData(JSONObject params) {
-		try {
-			String url = routes.get("api.candle.data");
-			JSONObject response = smartAPIRequestHandler.postRequest(this.apiKey, url, params, accessToken);
-			System.out.println(response);
-			return response.getString("data");
-		} catch (Exception | SmartAPIException e) {
-			System.out.println(e.getMessage());
-			return null;
-		}
-	}
-
-	/**
-	 * Logs out user by invalidating the access token.
-	 * 
-	 * @return JSONObject which contains status
-	 * 
-	 */
-
-	public JSONObject logout() {
-		try {
-			String url = routes.get("api.user.logout");
-			JSONObject params = new JSONObject();
-			params.put("clientcode", this.userId);
-			JSONObject response = smartAPIRequestHandler.postRequest(this.apiKey, url, params, accessToken);
-			return response;
-		} catch (Exception | SmartAPIException e) {
-			System.out.println(e.getMessage());
-			return null;
-		}
-	}
+
+    private static final Logger logger = LoggerFactory.getLogger(SmartConnect.class);
+    private static final Routes routes = new Routes();
+    private static final Proxy proxy = Proxy.NO_PROXY;
+    public static SessionExpiryHook sessionExpiryHook = null;
+    public static boolean enableLogging = false;
+    private SmartAPIRequestHandler smartAPIRequestHandler;
+    private String apiKey;
+    private String orderID = "orderid";
+    private String accessToken;
+    private String refreshToken;
+    private String userId;
+
+
+    public SmartConnect(String apiKey) {
+        this.apiKey = apiKey;
+    }
+
+    public SmartConnect(String apiKey, String accessToken, String refreshToken) {
+        this.apiKey = apiKey;
+        this.accessToken = accessToken;
+        this.refreshToken = refreshToken;
+    }
+
+
+    /**
+     * Returns apiKey of the App.
+     *
+     * @return String apiKey is returned.
+     * @throws CustomException if _apiKey is not found.
+     */
+    public String getApiKey() throws CustomException {
+        if (apiKey != null) return apiKey;
+        else throw new CustomException("The API key is missing.");

```

Comment:
Plase rename CustomException to a relevant name.

Reply:
have rename the class from CustomException to SmartConnectException